### ISSUE TITLE
Zero-shot text-to-image generation

### ISSUE THREAD

**Issue by jonghyunlee1993** [jonghyunlee1993/PaperArchive#3]
# DALL-E
- Discrete VAE 를 이용하여 이미지를 token화하고, 해당 이미지에 대한 caption을 training하여 이미지와 캡션 사이의 관계를 학습시킨다. 
- dVAE의 장점에 대해서 본 연구에서는 이러한 표현을 쓰고 있다. `high-frequency의 detail 보다는 low-frequency의 structure를 잘 유지할 수 있다.` 인지 심리학에서 [spatial frequency](https://en.wikipedia.org/wiki/Spatial_frequency)라는 표현을 사용하는데, 일맥상통하는 부분인 것으로 보인다. 
- 생성을 위해서 auto-regressive 한 Transformer decoder 구조를 쌓아올리고 causal mask를 통해 generation task를 학습한다. 
- 그리고 대량의 데이터와 다소 복잡한 GPU 최적화 기법 등을 이용해 이미지와 텍스트의 관계를 이해하는 모델을 생성하였다. 그래서 text 가이드를 통해 image 를 생성하는 것이 가능해졌다. 

# Loss function
![image](https://user-images.githubusercontent.com/37280722/183252608-97e07d36-55d6-4dcf-a141-629789d70861.png)
- $x$ 는 이미지, $y$는 캡션, $z$는 토큰으로 encoded된 RGB 이미지를 의미한다. 즉 캡션과 이미지 토큰을 가지고 $x$ 를 재생성하는 것이 목표
- 이 과정에서 $y$, $z$의 관계(joint distribution) 가 잘 학습될 것이다.

# 잡설
- (VAE regression)[https://arxiv.org/abs/1904.05948] 해당 페이퍼는 VAE 구조를 이용해서 brain age 를 regression하는 문제로 풀었는데, 여기에서 loss function 이 DALL-E의 loss function 과 닮아있다. DALL-E에서는 y를 텍스트를 이용한 caption으로 주었다면 VAE regression 은 연속형 변수인 나이로 주었다는 점 정도. 
- Text 와 Image의 loss 정도를 조절한 것 중 1/8:7/8로 조정한 부분. 아마 경험적으로 1/2 이랑은 차이가 있어야 하겠고, 1/4:3/4 해봤는데 성과가 그닥 좋지 않아서 경험적으로 설정한 것이 아닐까. 그리고 긴 텍스트에서 모든 부분이 다 정보가 있지는 않으니까 타당한 접근인 것 같다. 아무래도 이미지가 한 token에 더 많은 양의 정보가 압축되었을 것이니까. 
- Mixed precision training이 pytorch-lightning 에서는 precision=16 option 으로 해결된다. 세상이 편해졌다는 것을 새삼 느끼며 감사하다. (개인적으로 mixed precision을 애용하고 있다.)
